### PR TITLE
Deploy and discovery fixes

### PR DESCRIPTION
**Pull request by marcschier**
* Settings router crash on null
* Produce non overlapping port ranges
* Produce cidr address ranges that do not overlap
* Report configuration in reported properties
* Disable host network configuration on windows.
* Adding version number to engineering tool
* Fix updating existing deployment does not work.

### CODE REVIEW

File: common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs [hunk 1]
```diff
@@ -228,6 +265,47 @@ public sealed class AddressRange {
...
+        /// <param name="sb"></param>
+        /// <returns></returns>
+        private void AppendTo(StringBuilder sb) {
+            long start = Low;
+            long end = High;
+            var first = true;
+            while (end >= start) {
+                byte maxSize = 32;
+                while (maxSize > 0) {
+                    var mask = (long)(Math.Pow(2, 32) - Math.Pow(2, 32 - (maxSize - 1)));
```
Comment (mregen) [inline common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs]: >var mask = (long)(Math.Pow(2, 32) - Math.Pow(2, 32 - (maxSize - 1))); [](start = 20, length = 69)

mask = 0xffffffff and << 1 on each iteration should do the same?

File: common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs [hunk 2]
```diff
@@ -228,6 +265,47 @@ public sealed class AddressRange {
...
+                first = false;
+                sb.Append(ip);
+                sb.Append("/");
+                sb.Append(maxSize);
+                if (Nic != "custom") {
+                    sb.Append(" [");
+                    sb.Append(Nic);
+                    sb.Append("]");
+                }
+                start += (long)Math.Pow(2, 32 - maxSize);
```
Comment (mregen) [inline common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs]: >Math.Pow(2, 32 - maxSize) [](start = 31, length = 25)

1 << (32-maxsize)?

File: common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs [hunk 3]
```diff
@@ -228,6 +265,47 @@ public sealed class AddressRange {
...
+                byte maxSize = 32;
+                while (maxSize > 0) {
+                    var mask = (long)(Math.Pow(2, 32) - Math.Pow(2, 32 - (maxSize - 1)));
+                    var maskBase = start & mask;
+                    if (maskBase != start) {
+                        break;
+                    }
+                    maxSize--;
+                }
+                var x = Math.Log(end - start + 1) / Math.Log(2);
```
Comment (mregen) [inline common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs]: >x = Math.Log(end - start + 1) / Math.Log(2); [](start = 20, length = 44)

I bet there is a good way to this without log..

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubDiscovererDeployment.cs [hunk 4]
```diff
@@ -104,9 +104,9 @@ public sealed class IoTHubDiscovererDeployment : IHostProcess {
                 // Windows
                 createOptions = @"
                 {
-                    ""HostConfig"": {
-                        ""CapAdd"": [ ""NET_ADMIN"" ]
-                    }
+           //         ""HostConfig"": {
+           //             ""CapAdd"": [ ""NET_ADMIN"" ]
+           //         }
```
Comment (mregen) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubDiscovererDeployment.cs]: is this a forgotten comment?

Comment (marcschier) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubDiscovererDeployment.cs]: This broke overnight and caused windows discovery to fail.  It is unnecessary - I will remove it.

File: common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs (hunk 1 above)
Comment (marcschier) [reply common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs]: Fixed

File: common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs (hunk 3 above)
Comment (marcschier) [reply common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs]: Wont fix for now

File: common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs (hunk 2 above)
Comment (marcschier) [reply common/src/Microsoft.Azure.IIoT.Net/src/Models/AddressRange.cs]: Fixed

**Pull request merged**